### ISSUE TITLE
Overriding Function.prototype.toString

### ISSUE THREAD

**Issue by mashlol** [visjs/vis-network#191]
In the distributed js (found here https://unpkg.com/vis-network/standalone/umd/vis-network.js), there is some code that tries to override Function.prototype.toString:

```
var redefine = createCommonjsModule(function (module) {
var getInternalState = internalState.get;
var enforceInternalState = internalState.enforce;
var TEMPLATE = String(functionToString).split('toString');

shared('inspectSource', function (it) {
return functionToString.call(it);
});

(module.exports = function (O, key, value, options) {
var unsafe = options ? !!options.unsafe : false;
var simple = options ? !!options.enumerable : false;
var noTargetGet = options ? !!options.noTargetGet : false;
if (typeof value == 'function') {
    if (typeof key == 'string' && !has(value, 'name')) createNonEnumerableProperty(value, 'name', key);
    enforceInternalState(value).source = TEMPLATE.join(typeof key == 'string' ? key : '');
}
if (O === global_1) {
    if (simple) O[key] = value;
    else setGlobal(key, value);
    return;
} else if (!unsafe) {
    delete O[key];
} else if (!noTargetGet && O[key]) {
    simple = true;
}
if (simple) O[key] = value;
else createNonEnumerableProperty(O, key, value);
// add fake Function#toString for correct work wrapped methods / constructors with methods like LoDash isNative
})(Function.prototype, 'toString', function toString() {
return typeof this == 'function' && getInternalState(this).source || functionToString.call(this);
});
});
```

It also happens again much further down as well.

I've had to remove this code to get it to work in my codebase, as we prevent this. It seems wrong to me that it's trying to do this - is this coming from CommonJS?

**Comment by Thomaash** [comment]
Hi @mashlol,

do you detect pollution manually or do you have a tool for it? If you have a tool I'd like to incorporate it into our tests so that this doesn't happen again.

Thanks.

**Comment by mashlol** [comment]
Afaik it's just a runtime exception we throw, to prevent overriding prototype functions like Function.prototype.x or Object.prototype.x. But I am not an expert on this.

**Comment by Thomaash** [comment]
7 days went by and I think I know where everything comes from. The offenders are Vis Util, which will be mostly solved by https://github.com/visjs/vis-util/pull/61, and Vis Network, which will be mostly solved by #204 and Vis Util update.

Though it seems that Babel stubbornly creates two global variables `__core-js_shared__` and `regeneratorRuntime`. These are at least new, nothing preexisting is overwritten. Is this still a problem for you @mashlol or anyone else interested?

**Comment by mashlol** [comment]
Thanks for looking into this! I don't think the globals are an issue for us.
